lib/salus/scanners/report_pom_xml.rb

```diff
@@ -0,0 +1,48 @@
+require 'salus/scanners/base'
+
+# Report Java library usage
+
+module Salus::Scanners
+  class ReportPomXml < Base
+    UNKNOWN_VERSION = 'unknown'.freeze
+
+    def run
+      shell_return = run_shell("bin/parse_pom_xml #{@repository.path_to_repo}/pom.xml", chdir: nil)
+
+      if !shell_return.success?
+        report_error(shell_return.stderr)
+        return
+      end
+
+      dependencies = nil
+      begin
+        dependencies = JSON.parse(shell_return.stdout)
+      rescue JSON::ParserError
+        err_msg = "Could not parse JSON returned by bin/parse_pom_xml's stdout!"
+        report_stderr(err_msg)
+        report_error(err_msg)
+      end
+
+      dependencies.each do |dependency|
```

Comment:
`dependencies` is set to `nil` above.  It there is an exception, then it will stay `nil`, and then the `dependencies.each` will throw an exception.

Reply:
So I think you should `return` if `JSON.parse` fails.